### PR TITLE
Refactor and rubocop

### PR DESCRIPTION
**Pull request by dushan-madetech**
Changing class names and running rubocop on the code

### CODE REVIEW

**Comment by chubberlisk** [comment]
Do you think it's better to have `@winning_sets` as an instance variable or a class constant?

https://github.com/RobertM905/tic_tac_toe_sparring/blob/6002a056848b7f03c2fbd57d94b31f85b632e39c/lib/determine_outcome.rb#L6-L15

File: lib/determine_outcome.rb [hunk 1]
```diff
@@ -1,27 +1,29 @@
...
+      [6, 4, 2]
     ]
   end
 
   def execute
     board = @game_state_gateway.retrieve
-    check_for_winner(board)
+    winner = check_for_winner(board)
+    winner = 3 if check_for_draw(board) == 9
+    return winner
```
Comment (chubberlisk) [inline lib/determine_outcome.rb]: Unnecessary `return`. Line 22 can just be `winner`.

File: lib/determine_outcome.rb [hunk 2]
```diff
@@ -31,9 +33,13 @@ def check_for_winner(board)
     winner
   end
 
+  def check_for_draw(board)
+   board.count { |x| x==1 || x==2}
```
Comment (chubberlisk) [inline lib/determine_outcome.rb]: Some very tiny styling stuff. 

Currently:
```ruby
def check_for_draw(board)
 board.count { |x| x==1 || x==2}
end
```

Should be:
```ruby
def check_for_draw(board)
  board.count { |x| x == 1 || x == 2 }
end
```

**Comment by chubberlisk** [comment]
For your `MakeMove` use case, you have considered if a tile is out of bounds. Have you thought of all possible exceptions? Should it be possible for a player to make a move on a tile that already has a 1 or 2 on the board?

File: spec/acceptance/tic_tac_toe_spec.rb [hunk 3]
```diff
@@ -1,47 +1,57 @@
...
+    LoadGame.new(game_state_gateway: game_state_gateway)
   end
-  let(:modify_game_state) do
-    ModifyGameState.new(game_state_gateway: game_state_gateway)
+  let(:make_move) do
+    MakeMove.new(game_state_gateway: game_state_gateway)
+  end
+  
+  def given_a_new_game
+    game_state = [0, 0, 0, 0, 0, 0, 0, 0, 0]
```
Comment (yusufsheiqh) [inline spec/acceptance/tic_tac_toe_spec.rb]: Can initialise a new array by implementing either of the following:

```ruby
game_state = Array.new(9, 0)
```

```ruby
game_state = [0] * 9
```

I think it reads a lot nicer.

File: lib/determine_outcome.rb (hunk 1 above)
Comment (dushan-madetech) [reply lib/determine_outcome.rb]: removed it, thanks 😄

File: lib/determine_outcome.rb (hunk 2 above)
Comment (dushan-madetech) [reply lib/determine_outcome.rb]: corrected this, thanks 😄

**Comment by yusufsheiqh** [comment]
The 1, 2 and 3 assertions are a little confusing where 1 is when player one wins, 2 is when player 2 wins and 3 is when there is a draw(?). I assume refactoring is a later plan so the acceptance tests read better.

File: spec/acceptance/tic_tac_toe_spec.rb (hunk 3 above)
Comment (dushan-madetech) [reply spec/acceptance/tic_tac_toe_spec.rb]: You're right, I've changed the code to use the first suggestion, thanks 😄

**Comment by dushan-madetech** [comment]
> The 1, 2 and 3 assertions are a little confusing where 1 is when player one wins, 2 is when player 2 wins and 3 is when there is a draw(?). I assume refactoring is a later plan so the acceptance tests read better.

You are correct about the assertions and refactoring, will get this sorted soon enough

**Pull request merged**